hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/scm/node/TestDiskBalancerDuringDecommissionAndMaintenance.java

```diff
@@ -0,0 +1,200 @@
+      ((DatanodeInfo) dn).updateStorageReports(
+          HddsTestUtils.getRandomNodeReport(20, 1).getStorageReportList());
+    }
+  }
+
+  @AfterAll
+  public static void cleanup() throws Exception {
+    if (cluster != null) {
+      cluster.shutdown();
+    }
+  }
+
+  @Test
+  public void testDiskBalancerWithDecommissionAndMaintenanceNodes()
+      throws IOException, InterruptedException, TimeoutException {
+    List<HddsDatanodeService> dns = cluster.getHddsDatanodes();
+    DatanodeDetails dnToDecommission = dns.get(0).getDatanodeDetails();
+    DatanodeDetails dnToMaintenance = dns.get(1).getDatanodeDetails();
+
+    // Start disk balancer on all DNs
+    diskBalancerManager.startDiskBalancer(
+        Optional.of(10.0),
+        Optional.of(10L),
+        Optional.of(5),
+        Optional.of(false),
+        Optional.empty());
+
+    NodeManager nm = cluster.getStorageContainerManager().getScmNodeManager();
+
+    // Decommission DN1
+    scmClient.decommissionNodes(
+        Arrays.asList(getDNHostAndPort(dnToDecommission)), false);
+    waitForDnToReachOpState(nm, dnToDecommission, DECOMMISSIONING);
+
+    // Start maintenance on DN2
+    scmClient.startMaintenanceNodes(
+        Arrays.asList(getDNHostAndPort(dnToMaintenance)), 0, false);
+    waitForDnToReachOpState(nm, dnToMaintenance, ENTERING_MAINTENANCE);
+
+    //get diskBalancer report
+    List<HddsProtos.DatanodeDiskBalancerInfoProto> reportProtoList =
+        diskBalancerManager.getDiskBalancerReport(5,
+            ClientVersion.CURRENT_VERSION);
+
+    //get diskBalancer status
+    List<HddsProtos.DatanodeDiskBalancerInfoProto> statusProtoList =
+        diskBalancerManager.getDiskBalancerStatus(Optional.empty(),
+            Optional.empty(),
+            ClientVersion.CURRENT_VERSION);
+
+    // Verify that decommissioning and maintenance DN is not
+    // included in DiskBalancer report and status
+    boolean isDecommissionedDnInReport = reportProtoList.stream()
+        .anyMatch(proto -> proto.getNode().getUuid().
+            equals(dnToDecommission.getUuid().toString()));
+    boolean isMaintenanceDnInReport = reportProtoList.stream()
+        .anyMatch(proto -> proto.getNode().getUuid().
+            equals(dnToMaintenance.getUuid().toString()));
+    boolean isDecommissionedDnInStatus = statusProtoList.stream()
+        .anyMatch(proto -> proto.getNode().getUuid().
+            equals(dnToDecommission.getUuid().toString()));
+    boolean isMaintenanceDnInStatus = statusProtoList.stream()
+        .anyMatch(proto -> proto.getNode().getUuid().
+            equals(dnToMaintenance.getUuid().toString()));
+
+    // Assert that the decommissioned DN is not present in both report and status
+    assertFalse(isDecommissionedDnInReport);
+    assertFalse(isMaintenanceDnInReport);
+    assertFalse(isDecommissionedDnInStatus);
+    assertFalse(isMaintenanceDnInStatus);
+
+    LogCapturer dnStateChangeLog = LogCapturer.captureLogs(
+        DiskBalancerService.class);
```

Comment:
Move this LogCapturer ahead, old logs before LogCapturer object creation will not be included in LogCapturer's output.